src/Raven.Server/Documents/Handlers/StreamingHandler.cs

```diff
@@ -138,12 +138,12 @@ public async Task StreamQueryPost()
                 var properties = GetStringValuesQueryString("field", false);
                 var propertiesArray = properties.Count == 0 ? null : properties.ToArray();
                 
-                // set the exported file name
-                var fileName = query.Metadata.IsCollectionQuery ? query.Metadata.CollectionName + "_collection" : "query_result";
-                fileName = $"{Database.Name}_{fileName}";
+                // set the exported file name prefix
+                var fileNamePrefix = query.Metadata.IsCollectionQuery ? query.Metadata.CollectionName + "_collection" : "query_result";
+                fileNamePrefix = $"{Database.Name}_{fileNamePrefix}";
                 if (string.IsNullOrWhiteSpace(debug) == false)
                 {
-                    using (var writer = GetIndexEntriesQueryResultWriter(format, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileName))
+                    using (var writer = GetIndexEntriesQueryResultWriter(format, debug, HttpContext.Response, context, ResponseBodyStream(), propertiesArray, fileNamePrefix))
```

Comment:
The value of `debug` should to be checked here, not inside `GetIndexEntriesQueryResultWriter`. If we add another debug option we'll call a dedicated method to handle it here, not from `GetIndexEntriesQueryResultWriter`

Reply:
done